### PR TITLE
Add in descriptors for ReverseOneToOne and ForwardOneToOne

### PR DESCRIPTION
**Pull request by Chris7**
@benjiec This adds in descriptors for Django 1.11 for two conditions not previously handled.

### CODE REVIEW

File: curious/graph.py [hunk 1]
```diff
@@ -139,12 +143,17 @@ def get_related_objects(instances, filters=None):
       queryset = queryset.extra(select={INPUT_ATTR_PREFIX: '%s.%s' % (table, pk_field)})
 
     # reverse FK from instance to related objects with FK to the instance
-    elif type(rel_obj_descriptor) == ReverseManyToOneDescriptor:
-      rel_field = rel_obj_descriptor.rel.field
+    elif type(rel_obj_descriptor) in (ReverseManyToOneDescriptor, ReverseOneToOneDescriptor):
+      if type(rel_obj_descriptor) == ReverseOneToOneDescriptor:
+        rel_obj = rel_obj_descriptor.related
+      else:
+        rel_obj = rel_obj_descriptor.rel
```
Comment (myw) [inline curious/graph.py]: Might read clearer as an if else tertiary assignment expression

File: tests/curious_tests/models.py [hunk 2]
```diff
@@ -27,6 +27,15 @@ def authors(instances, filters=None):
     return r
 
 
+class UserTest(models.Model):
```
Comment (myw) [inline tests/curious_tests/models.py]: Not sure why this is called user here. Another noun from the existing world of authors, etc might be clearer. Or an explanation of why User is meaningful

File: tests/curious_tests/test_simple_queries.py [hunk 3]
```diff
@@ -76,7 +76,26 @@ def test_query_traversing_to_fk_object(self):
...
+    user_one_to_one = UserOneToOne(user=user_test, extra='extra')
+    user_one_to_one.save()
+
+    qs = 'UserOneToOne(extra="extra") UserOneToOne.user'
+    query = Query(qs)
+    result = query()
+    assertQueryResultsEqual(self, result[0][0][0], [(user_test, None)])
+    self.assertEquals(result[1], UserTest)
+
+    # go back
```
Comment (myw) [inline tests/curious_tests/test_simple_queries.py]: passing in self to an unbound method is ticky here. Csn this method be added via a mixin instead?

File: curious/graph.py (hunk 1 above)
Comment (myw) [reply curious/graph.py]: Also kinda a pain that the name is different in this confusing wat

Comment (Chris7) [reply curious/graph.py]: There are a ton of copy/paste blocks in this function that should just figure out how to pick out the related object. I agree it's annoying the descriptor has a different attribute -- that may actually be fixable if one is just there for compatibility. I'll swap to a ternary

Comment (Chris7) [reply curious/graph.py]: Ha no, they don't have a common method. I'm surprised about that, but this still is all in the `not meant for public consumption category of Django`

File: tests/curious_tests/models.py (hunk 2 above)
Comment (Chris7) [reply tests/curious_tests/models.py]: I couldn't figure out a legit One-To-One name for Authors last night. The canonical case is for extending the user object so I just made it User

File: tests/curious_tests/test_simple_queries.py (hunk 3 above)
Comment (Chris7) [reply tests/curious_tests/test_simple_queries.py]: It's a copy paste from how other test methods are done.

File: tests/curious_tests/models.py (hunk 2 above)
Comment (Chris7) [reply tests/curious_tests/models.py]: Author can now be a Person

File: tests/curious_tests/models.py [hunk 4]
```diff
@@ -27,11 +27,17 @@ def authors(instances, filters=None):
     return r
 
 
+class Person(models.Model):
+  gender = models.CharField(max_length=50)
```
Comment (myw) [inline tests/curious_tests/models.py]: the gender binary is a lie

**Pull request merged**